test/comments/updateComment.test.js

```diff
@@ -0,0 +1,282 @@
+import { ArticleData20 } from '../articles/__mocks__';
+
+const { Article } = models;
+const { rightUserWithEmail } = userData;
+
+chai.use(chaiHttp);
+
+let author;
+let artilceSlug;
+let commenter;
+let commenter2;
+let firstCommentId;
+
+const baseUrl = process.env.BASE_URL;
+
+before(async () => {
+  const user = getNewUser();
+  const signUpResponse = await chai
+    .request(app)
+    .post(`${baseUrl}/users/create`)
+    .send({
+      ...user,
+      confirmPassword: user.password
+    });
+  commenter = signUpResponse.body;
+
+  const user2 = getNewUser();
+  const signUpResponse2 = await chai
+    .request(app)
+    .post(`${baseUrl}/users/create`)
+    .send({
+      ...user2,
+      confirmPassword: user2.password
+    });
+  commenter2 = signUpResponse2.body;
+
+  const loginResponse = await chai
+    .request(app)
+    .post(`${baseUrl}/sessions/create`)
+    .send(rightUserWithEmail);
+  author = loginResponse.body;
+
+  const articleResponse = await chai
+    .request(app)
+    .post(`${baseUrl}/articles/create`)
+    .set('Authorization', author.token)
+    .send(ArticleData20);
+  artilceSlug = articleResponse.body.slug;
+
+  const res = await chai
+    .request(app)
+    .post(`${baseUrl}/articles/${artilceSlug}/comments`)
+    .set('authorization', commenter.token)
+    .send({
+      comment: new Array(4999).join('a')
+    });
+  firstCommentId = res.body.comment.id;
+});
+
+describe('Update comment test', async () => {
+  context(
+    'When an authenticated user updates comment with characters above 5000',
+    () => {
+      it('returns validation error', async () => {
+        const res = await chai
+          .request(app)
+          .patch(
+            `${baseUrl}/articles/${artilceSlug}/comments/${firstCommentId}`
+          )
+          .set('authorization', commenter.token)
+          .send({
+            comment: new Array(5555).join('a')
+          });
+        expect(res).to.have.status(422);
+        expect(res.body).to.haveOwnProperty('errors');
+        expect(res.body.errors).to.haveOwnProperty('comment');
+        expect(res.body.errors.comment).to.deep.equal(
+          'comment should not be more than 5000 characters'
+        );
+      });
+    }
+  );
+
+  context('When updating with comment id that is unavailable', () => {
+    it('returns error that comment does not exist', async () => {
+      const res = await chai
+        .request(app)
+        .patch(`${baseUrl}/articles/${artilceSlug}/comments/1000`)
+        .set('authorization', commenter.token)
+        .send({
+          comment: new Array(23).join('a')
+        });
+      expect(res).to.have.status(404);
+      expect(res.body).to.haveOwnProperty('error');
+      expect(res.body.error).to.deep.equal('comment not found');
+    });
+  });
+
+  context(
+    'When another authenticated user attempts to update another users comment',
+    () => {
+      it('returns error that comment does not exist', async () => {
```

Comment:
Why should the response be that the comment does not exist? This is an error with authorization. Could you return a 403 instead?